lib/storage/storage.ts

```diff
@@ -27,22 +30,37 @@ export class Storage extends BaseStorage implements IStorageInstance {
   }
 
   async delete(key: string, options?: Options) {
-    const result = await this.storageFetch<ErrorResponse>(key, {method: 'DELETE'}, options);
+    const result = await this.storageFetch<ErrorResponse>(key, { method: 'DELETE' }, options);
     if (result?.error) {
       return { error: result.error, success: false };
     } else {
       return { success: true };
     }
   }
-  
+
+  async search<T extends JsonValue>(key: string, options: SearchOptions = {}): Promise<SearchResponse<T>> {
+    const url = this.searchUrl(key, options);
+    const params = { method: 'GET' };
+    const result = await this.storageFetchV2<SearchServerResponse<T>>(url, params);
+    if (!isDefined(result)) {
+      return { success: false, records: null };
+    }
+
+    const response: SearchResponse<T> = {success: true, records: result.records};
```

Comment:
Your prettier/linter is working partially.
This creates a strange un-even formatting 😅